connect/runtime/src/main/java/org/apache/kafka/connect/runtime/AbstractHerder.java

```diff
@@ -210,6 +211,15 @@ public void onDeletion(ConnectorTaskId id) {
         statusBackingStore.put(new TaskStatus(id, TaskStatus.State.DESTROYED, workerId, generation()));
     }
 
+    public void onRestart(String connector) {
```

Comment:
I added the missing documentation.

I didn't add the onRestart method to  Listener because it didn't felt it should be a part of the task lifecycle controlled by the worker. Earlier the method was called recordRestart but based on a review comment, it was renamed it to onRestart, Do you think we should rename it back to recordRestart. The original intent of this method was to just record the state change.

Reply:
Did you compare it with other methods? My understanding is that `onDelete` is the same in that respect and exists on the `Listener` as well. But I might have skimmed too quickly through the code

Reply:
@kkonstantine you are right. Do you mind if we do this work in a follow-up PR?  The reason I ask is because if we add it to the listener then it becomes part of the interface and this would require me to retrofit the listener event into the old restartTask and restartConnector API for backward compatibility reasons and it can be big change to this already big PR.

Reply:
@kkonstantine I moved it to the Lisetner interface in https://github.com/apache/kafka/pull/10822/commits/7f8a588aaba9f1f80889e0429b2b9f495b1c666b.  Could you please check to see if it looks good.